src/main/kotlin/FilePathGenerator.kt

```diff
@@ -0,0 +1,103 @@
+import java.io.File
+import java.lang.IllegalArgumentException
+
+enum class SupportedExtensions(val ext: String) {
+    WAV("wav"),
+    MP3("mp3"),
+    JPEG("jpeg"),
+    JPG("jpg"),
+    TR("tr");
+
+    companion object {
+        private fun getSupportedContainers(): Array<SupportedExtensions> = arrayOf(TR)
+        private fun getCompressedTypes(): Array<SupportedExtensions> = arrayOf(MP3, JPEG, JPG)
```

Comment:
The concept of compressed/uncompressed types could be expressed as Enums. Enum classes can't extend an Enum, but you could make SupportedExtensions an interface and have both Enum types implement that interface; in which you can then use instanceOf (and/or the keywords 'is' and 'as') to handle determining if an enum is one or the other. That interface can also use the builder pattern, providing a function to take in your string and trying to construct the appropriate enum from it.